fs.go

```diff
@@ -143,3 +165,27 @@ type File interface {
 	// Truncate the file.
 	Truncate(size int64) error
 }
+
+// Capable interface can return the available features of a filesystem.
+type Capable interface {
+	// Capabilities returns the capabilities of a filesystem in bit flags.
+	Capabilities() Capability
+}
+
+// Capabilities returns the features supported by a filesystem. If the FS
+// does not implement Capable interface it returns all features.
+func Capabilities(fs Basic) Capability {
```

Comment:
In order to maintain backwards compatibility, this method should assume a fixed set of capabilities for filesystems not implementing `Capable`, not something that can evolve over time.

Maybe we could rename `CapAll` to `CapDefault` and state explicitly in the documentation that `CapDefault` values will never change without a major version bump.

Reply:
My comment is silly and very personal :) but just adding 2 cents... 
For me `Cap` sounds like _Capacity_ (especially that we have in go global function `cap` which returns capacity). So maybe we can rename `Capability` to `Ability` (or `FSMode`) and consts to `Readable`, `Writable`, `Seekable`, `Lockable`, ...

Reply:
There is something like this for the interfaces implemented by the FS. This implementation may not be very Goish. It tries to mimic open flags (`O_RDONLY | O_SYNC`) for operations that are not implemented in the billy FS even if the methods are there. One example is go-billy-siva. The files can not be opened in read and write mode. We can check with this flags before using one algorithm or the other:

```
if billy.CapabilityCheck(mySivaFS, billy.ReadAndWrite) {
....
}
```

Other example is memory FS. The locking methods are no-ops.

`Capability` may be changed to `Feature` if it seems too close to capacity.

Reply:
Feature sounds good to me. But I'm not totally against capability, but just abbreviation `cap` looks more like _capacity_.

Reply:
Using `Capability` prefix instead of `Cap` also looks more in line with our convention of avoiding abbreviations, specially if there is any ambiguity.